src/providers/aws/mod.rs

```diff
@@ -49,7 +50,18 @@ pub struct AwsProvider {
 
 impl AwsProvider {
     pub fn try_new() -> Result<AwsProvider> {
-        let client = retry::Client::try_new()?.return_on_404(true);
+        let mut client = retry::Client::try_new()?.return_on_404(true);
+        let token = AwsProvider::fetch_imdsv2_token(client.clone());
+
+        // If IMDSv2 token is fetched successfully, set the header.
+        // Otherwise, proceed with IMDSv1 mechanism.
+        if let Ok(t) = token {
```

Comment:
I'd rather turn this into a full `match` and perform some kind of `log::warn!()` in the `Err` arm.

If I've correctly understood the docs, the only way for this to fail is when the metadata is disabled/firewalled completely. If that is the case, having a warning logged here would help detecting fishy setups.

Reply:
I believe that is the case, will update with `log::warn!()`

Reply:
Well I think my original logic failed to catch when `token` is an error and still returned `Ok` even if failed to fetch token.